docs/advanced/pycpp/object.rst

```diff
@@ -176,6 +179,47 @@ Generalized unpacking according to PEP448_ is also supported:
 
 .. _PEP448: https://www.python.org/dev/peps/pep-0448/
 
+.. _implicit_casting:
+
+Implicit casting
+================
+
+When using this C++ interface for Python types or calling Python functions,
+and objects of type :class:`object` are returned, it is possible to
+specialize to a subtype like :class:`dict`. The same holds for the proxy objects
```

Comment:
The first sentence is too long and somewhat difficult to read. I think this would be nicer to read.

When using the C++ interface for Python types, or calling Python functions, objects of type :class:`object` are returned. It is possible to invoke implicit conversions to a subclass like :class:`dict`.

Usually, I don't pay much attention to the wording of the docs, but this time it actually took me 3 readings to figure out what is this trying to convey.

Reply:
Thanks! That's definitely easier to read. I've adapted the text.